hypershift-operator/controllers/nodepool/metrics/metrics.go

```diff
@@ -34,6 +35,13 @@ var (
 		Help: "Time in seconds it took from initial NodePool creation and rollout of initial version",
 		Name: NodePoolInitialRolloutDurationMetricName,
 	}, labelNames)
+
+	NodePoolTransitionSecondsMetricName = "hypershift_nodepools_transition_seconds"
+	nodePoolTransitionSeconds           = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    NodePoolTransitionSecondsMetricName,
+		Help:    "Time in seconds it took from NodePool creation until a given condition transitions to true",
```

Comment:
Do we know the "given condition"?  Could this be explained better?

Reply:
we are tracking multiple conditions with this metric, each condition will have its own time series with the condition name/type set as a label.
the help won't show in Grafana UI as far as I know.